tests/rptest/tests/cluster_metrics_test.py

```diff
@@ -93,11 +83,34 @@ def _get_cluster_metrics(
             self, node: ClusterNode) -> Optional[dict[str, MetricSamples]]:
         """
         Get all the cluster level metrics exposed by a specified
-        node in the cluster.
-        """
+        node in the cluster. If 'node' is the current controller,
+        wait briefly for the metrics to become available.
+        """
+        def get_metrics_from_node_sync(pattern: str):
+            samples = self.redpanda.metrics_sample(
+                pattern, [node], MetricsEndpoint.PUBLIC_METRICS)
+            success = samples is not None
+            return success, samples
+
+        def get_metrics_from_node_async(pattern: str):
+            metrics = None
+            try:
+                metrics = wait_until_result(
+                    lambda: get_metrics_from_node_sync(pattern),
+                    timeout_sec=2,
+                    backoff_sec=.1)
+            except TimeoutError:
+                pass
+
+            return metrics
+
         def get_metrics_from_node(pattern: str):
-            return self.redpanda.metrics_sample(pattern, [node],
-                                                MetricsEndpoint.PUBLIC_METRICS)
+            if node == self.redpanda.controller():
```

Comment:
when we invoke `_get_cluster_metrics` we already know whether or not the node should have data so let's add a boolean `expect_metrics` argument to the method instead of deducing it from `self.redpanda.controller()`. It's more robust.

Reply:
I agree. Good suggestion.